core/src/main/java/com/opensymphony/xwork2/ActionContext.java

```diff
@@ -185,6 +187,10 @@ public void setContextMap(Map<String, Object> contextMap) {
      * @return the context map.
      */
     public Map<String, Object> getContextMap() {
+        Map<String, Object> context = getContext().context;
+        if (context instanceof OgnlContext) {
+            ((OgnlContext) context).put(TypeConverter.TYPE_CONVERTER_CONTEXT_KEY, ((OgnlContext) context).getTypeConverter());
```

Comment:
What is the deal with the type converter?

Reply:
Good question, probably it's a workaround :/ Need to re-think this.

Reply:
Whooops... it can be thrown away, goot catch 👍